### PR TITLE
Hyper Client

### PR DESCRIPTION
**Pull request by developeruche**
Introduces the hyperclient, a wasm-compatible rust library for managing in-flight cross-chain requests & responses

### CODE REVIEW

File: client/src/providers/evm_chain.rs [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+    }
+}
+
+
+impl Client for EvmClient {
+    async fn query_latest_block_height(&self) -> Result<u64, anyhow::Error> {
+        Ok(self.client.get_block_number().await?.as_u64())
+    }
+
+    async fn query_latest_block_timestamp(&self) -> Result<u64, anyhow::Error> {
```
Comment (Wizdave97) [inline client/src/providers/evm_chain.rs:83]: This is a duplicate of `host_timestamp`.

File: client/src/providers/hyperbridge.rs [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    }
+
+    pub async fn query_request(
+        &self,
+        source_chain: &StateMachine,
+        dest_chain: &StateMachine,
+        nonce: u64,
+    ) -> Result<Vec<Request>, anyhow::Error> {
+        let build_leaf_index_query =
+            LeafIndexQuery { source_chain: *source_chain, dest_chain: *dest_chain, nonce };
```
Comment (Wizdave97) [inline client/src/providers/hyperbridge.rs:53]: The only field in LeafIndexQuery is just the request or response commitment, pull changes from main

File: client/src/providers/hyperbridge.rs [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+            self.client.rpc().request("ismp_queryResponses", leaf_index_query).await?;
+
+        Ok(hyper_bridge_response)
+    }
+
+    pub async fn query_state_machine_commitment(
+        &self,
+        height: StateMachineHeight,
+    ) -> Result<StateCommitment, anyhow::Error> {
+        let mut partial_key =
```
Comment (Wizdave97) [inline client/src/providers/hyperbridge.rs:82]: Use the subxt generated types to read storage.

File: client/src/providers/hyperbridge.rs [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+    }
+}
+
+const REQUEST_PARTIAL_KEY: [u8; 32] =
+    hex!("103895530afb23bb607661426d55eb8bbd3caa596ab5c98b359f0ffc7d17e376");
+const RESPONSE_PARTIAL_KEY: [u8; 32] =
+    hex!("103895530afb23bb607661426d55eb8b8fdfbc1b10c58ed36779810ffdba8e79");
+
+pub fn get_request_storage_key(request_commitment: Vec<u8>) -> Vec<u8> {
+    let mut key = REQUEST_PARTIAL_KEY.to_vec();
```
Comment (Wizdave97) [inline client/src/providers/hyperbridge.rs:145]: Use subxt generated types to get storage keys

**Review by Wizdave97 — CHANGES_REQUESTED** [review]

File: client/src/providers/hyperbridge.rs (hunk 3 above)
Comment (developeruche) [reply client/src/providers/hyperbridge.rs:82]: Please could you explain this further, in the `runtime.rs` file, I could find a function that could replace this function.

Comment (Wizdave97) [reply client/src/providers/hyperbridge.rs:78]: Check how we read storage items in tesseract-substrate calls.rs

File: client/pkg/hyperclient.d.ts [hunk 5]
```diff
@@ -0,0 +1,89 @@
+/* tslint:disable */
+/* eslint-disable */
+/**
+* Functions takes in a post request and returns a `MessageStatus`
+* @param {any} request
+* @param {any} config_js
+* @returns {Promise<any>}
+*/
+export function query_request_status(request: any, config_js: any): Promise<any>;
```
Comment (seunlanlege) [inline client/pkg/hyperclient.d.ts:9]: Would be nice to have the actual type definitions

**Review by seunlanlege — APPROVED** [review]

**Pull request merged**